### PR TITLE
Update Azure SDK & Service API versions

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/9165

Update Azure SDK and Service API, since we were using old versions.

### CODE REVIEW

**Comment by fiunchinho** [comment]
E2E tests passed correctly.

I also deployed this branch on a test installation and manually created a test to run some tests. Everything seems to work normally.

File: client/azure_client_set.go [hunk 1]
```diff
@@ -3,9 +3,9 @@ package client
 import (
 	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
 	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2017-10-01/dns"
```
Comment (nprokopic) [inline client/azure_client_set.go]: What about other APIs, for example DNS here? Any specific reason not to update them?

There is a newer Service API version https://github.com/Azure/azure-sdk-for-go/tree/master/services/dns/mgmt/2018-05-01/dns

Comment (fiunchinho) [reply client/azure_client_set.go]: I was focusing on the ones that could yield bigger impact. Should I update all of them? Maybe that's a good idea.

**Comment by fiunchinho** [comment]
Done. I'll re-run e2e tests

File: client/azure_client_set.go (hunk 1 above)
Comment (nprokopic) [reply client/azure_client_set.go]: I would go for all of them :)

**Comment by fiunchinho** [comment]
Manual test was ok. Sonobuoy tests were ok.

**Pull request merged**